File: memorizingpager/src/main/java/com/coloredpanda/memorizingpager/MotionlessPager.java
package com.coloredpanda.memorizingpager;

import android.content.Context;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.util.Log;
import android.view.MotionEvent;
import android.view.animation.DecelerateInterpolator;
import android.widget.Scroller;

import java.lang.reflect.Field;

/**
 * MotionlessPager it class that allows to disable paging by swiping of {@link ViewPager}
 *
 * @version 1.0
 */

public abstract class MotionlessPager extends ViewPager {

    private static final String TAG = "MotionlessPager";

    public MotionlessPager(Context context) {
        super(context);
        init();
    }

    public MotionlessPager(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    private void init() {
        try {
            Class<?> viewpager = ViewPager.class;
            Field scrollerField = viewpager.getDeclaredField("mScroller");
            scrollerField.setAccessible(true);
            scrollerField.set(this, new MotionlessScroller(getContext()));
        } catch (Exception e) {
            Log.e(TAG, e.getMessage());
        }
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent event) {
        return false;
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        return false;
    }

    private static final class MotionlessScroller extends Scroller {

        MotionlessScroller(Context context) {
            super(context, new DecelerateInterpolator());
        }

        @Override
        public void startScroll(int startX, int startY, int dx, int dy, int duration) {
            super.startScroll(startX, startY, dx, dy, 0);
        }

    }

}

File: memorizingpager/src/main/java/com/coloredpanda/memorizingpager/MotionlessPager.kt
package com.coloredpanda.memorizingpager

import android.content.Context
import android.support.v4.view.ViewPager
import android.util.AttributeSet
import android.util.Log
import android.view.MotionEvent
import android.view.animation.DecelerateInterpolator
import android.widget.Scroller

import java.lang.reflect.Field

/**
 * MotionlessPager it class that allows to disable paging by swiping of [ViewPager]
 *
 * @version 1.0
 */

abstract class MotionlessPager : ViewPager {

    constructor(context: Context) : super(context) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init()
    }

    private fun init() {
        try {
            val viewpager = ViewPager::class.java
            val scrollerField = viewpager.getDeclaredField("mScroller")
            scrollerField.isAccessible = true
            scrollerField.set(this, MotionlessScroller(context))
        } catch (e: Exception) {
            Log.e(TAG, e.message)
        }

    }

    override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
        return false
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        return false
    }

    private class MotionlessScroller internal constructor(context: Context) : Scroller(context, DecelerateInterpolator()) {

        override fun startScroll(startX: Int, startY: Int, dx: Int, dy: Int, duration: Int) {
            super.startScroll(startX, startY, dx, dy, 0)
        }

    }

    companion object {

        private val TAG = "MotionlessPager"
    }

}
